stress: failed test in cockroach/storage/storage.test: TestReplicaLeaseCounters

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/2f89e7860ada6f76e253481b519c95b647ce564a

Stress build found a failed test:

```
=== RUN   TestReplicaLeaseCounters
W160430 07:17:14.257328 gossip/gossip.go:892  not connected to cluster; use --join to specify a connected node
I160430 07:17:14.257384 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160430 07:17:14.259213 storage/replica_command.go:1440  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 1s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=1970-01-01 00:00:00 +0000 UTC]
I160430 07:17:14.259916 stopper.go:352  draining; tasks left:
3      gossip/infostore.go:286
I160430 07:17:14.259963 stopper.go:352  draining; tasks left:
2      gossip/infostore.go:286
I160430 07:17:14.259986 stopper.go:352  draining; tasks left:
1      gossip/infostore.go:286
--- FAIL: TestReplicaLeaseCounters (0.01s)
	replica_test.go:712: metrics counters actual=2, expected=1
```

Run Details:

```
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s
0 runs so far, 0 failures, over 15s
0 runs so far, 0 failures, over 20s
1 runs so far, 0 failures, over 25s
15 runs so far, 0 failures, over 30s
16 runs so far, 0 failures, over 35s
16 runs so far, 0 failures, over 40s
16 runs so far, 0 failures, over 45s
17 runs so far, 0 failures, over 50s
30 runs so far, 0 failures, over 55s
32 runs so far, 0 failures, over 1m0s
32 runs so far, 0 failures, over 1m5s
32 runs so far, 0 failures, over 1m10s
33 runs so far, 0 failures, over 1m15s
38 runs completed, 1 failures, over 1m20s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/storage/replica_test.go
+++ b/storage/replica_test.go
@@ -703,17 +703,30 @@ func TestReplicaNotLeaderError(t *testing.T) {
 // correctly after a lease request.
 func TestReplicaLeaseCounters(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+
+	var numProp int64
+	sCtx := TestStoreContext()
+	sCtx.TestingKnobs.TestingCommandFilter = func(
+		args storageutils.FilterArgs,
+	) *roachpb.Error {
+		if args.Req.Method() == roachpb.LeaderLease {
+			atomic.AddInt64(&numProp, 1)
+		}
+		return nil
+	}
+
 	tc := testContext{}
-	tc.Start(t)
+	tc.StartWithStoreContext(t, sCtx)
 	defer tc.Stop()
 
 	assert := func(actual, expected int64) {
 		if actual != expected {
-			t.Fatalf("metrics counters actual=%d, expected=%d", actual, expected)
+			t.Fatal(util.ErrorfSkipFrames(1,
+				"metrics counters actual=%d, expected=%d", actual, expected))
 		}
 	}
 	metrics := tc.rng.store.metrics
-	assert(metrics.leaseRequestSuccessCount.Count(), 1)
+	assert(metrics.leaseRequestSuccessCount.Count(), atomic.LoadInt64(&numProp))
 	assert(metrics.leaseRequestErrorCount.Count(), 0)
 
 	now := tc.clock.Now()
@@ -729,7 +742,8 @@ func TestReplicaLeaseCounters(t *testing.T) {
 	}); err != nil {
 		t.Fatal(err)
 	}
-	assert(metrics.leaseRequestSuccessCount.Count(), 2)
+	lastCount := atomic.LoadInt64(&numProp)
+	assert(metrics.leaseRequestSuccessCount.Count(), lastCount)
 	assert(metrics.leaseRequestErrorCount.Count(), 0)
 
 	// Make setLeaderLease fail by providing an invalid ReplicaDescriptor.
@@ -743,11 +757,11 @@ func TestReplicaLeaseCounters(t *testing.T) {
 			StoreID:   99,
 		},
 	}); err == nil {
-		t.Fatal("Expected setLeaderLease to fail on invalid ReplicaDescriptor. It didn't.")
+		t.Fatal("setLeaderLease did not fail on invalid ReplicaDescriptor")
 	}
 
-	assert(metrics.leaseRequestSuccessCount.Count(), 2)
-	assert(metrics.leaseRequestErrorCount.Count(), 1)
+	assert(metrics.leaseRequestSuccessCount.Count(), lastCount)
+	assert(metrics.leaseRequestErrorCount.Count(), atomic.LoadInt64(&numProp)-lastCount)
 }
 
 // TestReplicaGossipConfigsOnLease verifies that config info is gossiped

